service/mantle/ups/UpsServices.xml

```diff
@@ -47,4 +47,137 @@
+        </out-parameters>
+        <actions>
+
+            <!-- Set shippingGatewayConfigId given in input else take value stored in routeSegment entity -->
+            <set field="shippingGatewayConfigId" from="shippingGatewayConfigId?:routeSegment.shippingGatewayConfigId"/>
+
+            <set field="carrierPartyId" from="carrierPartyId ?: routeSegment.carrierPartyId"/>
+            <if condition="!carrierPartyId || carrierPartyId == '_NA_'">
+                <return message="Carrier not set on ${shipmentId}:${shipmentRouteSegmentSeqId}"/>
+            </if>
+
+            <!-- Get ShipmentMethodTypeEnumId from the input or from routeSegment entity-->
+            <set field="shipmentMethodEnumId" from="shipmentMethodEnumId?:routeSegment.shipmentMethodEnumId"/>
+            <if condition="!shipmentMethodEnumId">
+                <return type="danger" message="In get#ShippingLabel Service, no shipmentMethod available."/>
+            </if>
+
+            <set field="hasIssues" from="false"/>
+            <set field="fromPartyId" from="shipment.fromPartyId"/>
+            <if condition="!fromPartyId">
+                <set field="hasIssues" from="true"/>
+                <message type="danger">From party not set on shipment ${shipmentId}</message>
+            </if>
+            <set field="toPartyId" from="shipment.toPartyId"/>
+            <if condition="!toPartyId">
+                <set field="hasIssues" from="true"/>
+                <message type="danger">To party not set on shipment ${shipmentId}</message>
+            </if>
+            <if condition="hasIssues">
+                <message type="danger">Issues found, not requesting Shippo labels</message>
+                <return/>
+            </if>
+
+            <!--address_to-->
+            <set field="destPostalContactMech" from="routeSegment.destPostalContactMechId"/>
+            <!-- If ShipmentRouteSegment Entity has no value of destPostalContactMechId, find contactMechId from PartyContactMech -->
+            <if condition="!destPostalContactMech">
+                <entity-find-one entity-name="mantle.party.contact.PartyContactMech" value-field="partyContactMech">
+                    <field-map field-name="partyId" from="toPartyId"/>
+                    <field-map field-name="contactMechPurposeId" value="PostalShippingDest"/>
+                </entity-find-one>
+                <set field="destPostalContactMech" from="partyContactMech.contactMechId"/>
+            </if>
+
+            <!-- call service for validating Shipping Destination Address -->
+            <service-call name="mantle.ups.UpsServices.validate#PostalAddress" out-map="destAddrOut" in-map="[contactMechId:destPostalContactMech, partyId:toPartyId, shippingGatewayConfigId:shippingGatewayConfigId]"/>
+
+            <!-- After validating dest address service will return the destAddrOut containing name and address of the receiver -->
+            <set field="nameTo" from="destAddrOut.address.Name"/>
```

Comment:
I think this can be potential NPE when address validation fails

Reply:
I think now there is no need to check NPE here as the code is already returning the service if the responseStatusCode is 0. In that case we'll always get address